src/main/java/jdbchomework/dao/jdbc/AbstractDao.java

```diff
@@ -2,118 +2,116 @@
 
 import jdbchomework.dao.model.GenericDao;
 import jdbchomework.entity.AbstractEntity;
-import org.hibernate.HibernateException;
-import org.hibernate.Session;
-import org.hibernate.SessionFactory;
-import org.hibernate.Transaction;
 import org.slf4j.LoggerFactory;
 
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
 import java.util.ArrayList;
 import java.util.List;
 
 public abstract class AbstractDao<T extends AbstractEntity> implements GenericDao<T> {
 
-    public static final String CANNOT_CONNECT_TO_DB = "Cannot connect to DB";
-    private org.slf4j.Logger log = LoggerFactory.getLogger(AbstractDao.class);
-    private SessionFactory sessionFactory;
-    private String entity;
-    private Class<T> clazz;
+    private static org.slf4j.Logger log = LoggerFactory.getLogger(AbstractDao.class);
+    private String table;
+    private String column;
 
-    public AbstractDao(SessionFactory sessionFactory, String entity, Class<T> clazz) {
-        this.sessionFactory = sessionFactory;
-        this.entity = entity;
-        this.clazz = clazz;
+    protected Connection connection;
+
+    public AbstractDao(Connection connection, String table, String column) {
+        this.connection = connection;
+        this.table = table;
+        this.column = column;
+    }
+
+    @Override
+    public boolean deleteById(long id) {
+        boolean result = false;
+        try (PreparedStatement statement = connection
+                .prepareStatement("DELETE FROM " + table + " WHERE " + column + " = ?;")) {
+            statement.setLong(1, id);
+            if (statement.executeUpdate() > 0) {
+                result = true;
+            }
+        } catch (SQLException e) {
+            log.error("Cannot connect to DB", e);
+            throw new RuntimeException(e);
+        }
+        return result;
     }
 
     public void add(T toAdd) {
-        Session session = getSession();
-        Transaction transaction = null;
-        try {
-            transaction = session.beginTransaction();
-            session.save(toAdd);
-            transaction.commit();
-        } catch (HibernateException e) {
-            if (transaction != null) {
-                transaction.rollback();
+        try (PreparedStatement statement = connection.prepareStatement("INSERT  INTO " + table + " (name)VALUES (?);")) {
+            connection.setTransactionIsolation(Connection.TRANSACTION_READ_COMMITTED);
+            connection.setAutoCommit(false);
+            statement.setString(1, toAdd.getName());
+            connection.commit();
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            log.error("Cannot connect to DB", e);
+            throw new RuntimeException("Cannot connect to DB", e);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)